packages/wrangler/src/r2/lifecycle.ts

```diff
@@ -0,0 +1,412 @@
+) {
+	await printWranglerBanner();
+	const config = readConfig(args.config, args);
+	const accountId = await requireAuth(config);
+
+	const { bucket, jurisdiction } = args;
+
+	logger.log(`Listing lifecycle rules for bucket '${bucket}'...`);
+
+	const lifecycleRules = await getLifecycleRules(
+		accountId,
+		bucket,
+		jurisdiction
+	);
+
+	if (lifecycleRules.length === 0) {
+		logger.log(`There are no lifecycle rules for bucket '${bucket}'.`);
+	} else {
+		const tableOutput = tableFromLifecycleRulesResponse(lifecycleRules);
+		logger.log(tableOutput.map((x) => formatLabelledValues(x)).join("\n\n"));
+	}
+}
+
+export function AddOptions(yargs: CommonYargsArgv) {
+	return yargs
+		.positional("bucket", {
+			describe: "The name of the R2 bucket to add a lifecycle rule to",
+			type: "string",
+			demandOption: true,
+		})
+		.positional("id", {
+			describe: "A unique identifier for the lifecycle rule",
+			type: "string",
+			requiresArg: true,
+		})
+		.positional("prefix", {
+			describe:
+				"Prefix condition for the lifecycle rule (leave empty for all prefixes)",
+			type: "string",
+			requiresArg: true,
+		})
+		.option("expire-days", {
+			describe: "Number of days after which objects expire",
+			type: "number",
+			requiresArg: true,
+		})
+		.option("expire-date", {
+			describe: "Date after which objects expire (YYYY-MM-DD)",
+			type: "number",
+			requiresArg: true,
+		})
+		.option("ia-transition-days", {
+			describe:
+				"Number of days after which objects transition to Infrequent Access storage",
+			type: "number",
+			requiresArg: true,
+		})
+		.option("ia-transition-date", {
+			describe:
+				"Date after which objects transition to Infrequent Access storage (YYYY-MM-DD)",
+			type: "string",
+			requiresArg: true,
+		})
+		.option("abort-multipart-days", {
+			describe:
+				"Number of days after which incomplete multipart uploads are aborted",
+			type: "number",
+			requiresArg: true,
+		})
+		.option("jurisdiction", {
+			describe: "The jurisdiction where the bucket exists",
+			alias: "J",
+			requiresArg: true,
+			type: "string",
+		});
+}
+
+export async function AddHandler(
+	args: StrictYargsOptionsToInterface<typeof AddOptions>
+) {
+	await printWranglerBanner();
+	const config = readConfig(args.config, args);
+	const accountId = await requireAuth(config);
+
+	const {
+		bucket,
+		expireDays,
+		expireDate,
+		iaTransitionDays,
+		iaTransitionDate,
+		abortMultipartDays,
+		jurisdiction,
+	} = args;
+
+	let { id, prefix } = args;
+
+	const isInteractive = process.stdin.isTTY;
+
+	const lifecycleRules = await getLifecycleRules(
+		accountId,
+		bucket,
+		jurisdiction
+	);
+
+	if (!id && isInteractive) {
+		id = await prompt("Enter a unique identifier for the lifecycle rule");
+	}
+
+	if (!id) {
+		throw new UserError("Must specify a rule ID.");
+	}
+
+	const newRule: LifecycleRule = {
+		id: id,
+		enabled: true,
+		conditions: {},
+	};
+
+	if (!prefix && isInteractive) {
```

Comment:
Do you need to differentiate whether the user have specified a prefix or not?
```suggestion
	if (prefix === undefined && isInteractive) {
```